File: 240519/완전수/perfect-number.py
start, end = input().split()
start = int(start); end = int(end)

cnt = 0

for i in range(start, end+1):
    sum_ = 0

    for j in range(1, i):
        if i%j == 0:
            sum_ += j
    if i == sum_:
        cnt += 1
print(cnt)

File: 241223/완전수/perfect-number.cpp
#include <iostream>
using namespace std;

int main() {
    // 여기에 코드를 작성해주세요.
    int start, end, cnt = 0;
    cin >> start >> end;

    for (; start<=end; start++) {
        int sum_val = 0;
        for (int i=1; i<start; i++) {
            if (start%i == 0) {
                sum_val += i;
            }
        }
        if (sum_val == start) {
            cnt++;
        }
    }
    cout << cnt;
    return 0;
}